### DISCUSSION TITLE
Add Feature to Play Audio Files

### DISCUSSION THREAD

**Discussion by ghost** [TagStudioDev/TagStudio#395] [Ideas]
Hello!

I think TagStudio's tag system would be a really efficient way of organizing my music library. It seems like organizing music by tags would be a lot more flexible/powerful than just making playlists. I realize that you already have the ability to tag audio files, but no ability to play them.

It might be outside the scope of this project, but what are thoughts about adding a feature to play audio files? If it is out of scope, would it be possible to develop a separate audio player that can read the tag database?

I would appreciate other ideas if anyone is interested.

**Comment by Qronikarz** [comment]
> It seems like organizing music by tags would be a lot more flexible/powerful than just making playlists.

Seems is an understatement. It is more powerful. At least for me.

> If it is out of scope, would it be possible to develop a separate audio player that can read the tag database?
I would appreciate other ideas if anyone is interested.

There is one other idea. I'm working on something like that for my own project. That is exporting the mp3 playlist to use with external player. I mean, I do have access to audio player in JavaScript but I found the concept of exporting the playlist for other programs very convenient and fun feature.

Here's the playlist format that can be made just with text manipulation: https://en.wikipedia.org/wiki/XML_Shareable_Playlist_Format
Supports windows paths, linux paths, relative paths and http locations.

The way it could work is you make your search query in TagStudio by selecting only mp3 files, using whatever tags you want and clicking the option to export the search results to XSPF format.

Most of the programs I use do support that format - AIMP, Audacious, VLC.

**Reply by ghost** [reply]
Oh wow, I didn't know about XSPF. I think it could be a good solution. Definitely simpler than making a built-in player.

If there is more interest in this, I would be more than willing to help out.
